zone/attack.cpp

```diff
@@ -5809,6 +5812,252 @@ const DamageTable &Mob::GetDamageTable() const
 	return which[level - 50];
 }
 
+int Mob::GetMobFixedOffenseSkill()
+{
+	// Due to new code using a combination of Offense and Weapon skill to determine hit, depending on the class
+	// and weapon wielded by a mob, the hit rate of an equal level mob could vary between 15% and 60%, which made
+	// many mobs far too easy.  This particular call replaces the class based Offense Skill with a fixed value
+	// equal to that of a Warrior of appropriate Level if UseMobFixedOffenseSkill flag is TRUE.
+
+	int level = std::max(1,static_cast<int>(GetLevel()));
+
+	// Current tables are flat above Level 60
+	if (level > 60) {
+		level = 60;
+	}
+
+	int FixedOffenseSkillTable[] = {
+			10, // 1
+			15,
+			20,
+			25,
+			30, // 5
+			35,
+			40,
+			45,
+			50,
+			55, // 10
+			60,
+			65,
+			70,
+			75,
+			80, // 15
+			85,
+			90,
+			95,
+			100,
+			105, // 20
+			110,
+			115,
+			120,
+			125,
+			130, // 25
+			135,
+			140,
+			145,
+			150,
+			155, // 30
+			160,
+			165,
+			170,
+			175,
+			180, // 35
+			185,
+			190,
+			195,
+			200,
+			205, // 40
+			210,
+			210,
+			210,
+			210,
+			210, // 45
+			210,
+			210,
+			210,
+			210,
+			210, // 50
+			215,
+			220,
+			225,
+			230,
+			235, // 55
+			240,
+			245,
+			250,
+			252,
+			252, // 60
+	};
+
+	return FixedOffenseSkillTable[level - 1];
+}
+
+int Mob::GetMobFixedWeaponSkill()
+{
+	// Due to new code using a combination of Offense and Weapon skill to determine hit, depending on the class
+	// and weapon wielded by a mob, the hit rate of an equal level mob could vary between 15% and 60%, which made
+	// many mobs far too easy.  This particular call replaces the weapon/class based Weapon Skill with a fixed value.
+	// Two tables exist, one equal to a Warrior of appropriate level, and one modified to make hit rate equal to the old code
+	// assuming the UseMobFixedOffenseSkill flag is set TRUE or the mob class is a Warrior (all the the bonus is in Weapon Skill).
+
+	int level = std::max(1,static_cast<int>(GetLevel()));
+
+	// Current tables are flat above Level 70
+	if (level > 70) {
+		level = 70;
+	}
+
+	int FixedWeaponSkillTable[] = {
+			10, // 1
+			15,
+			20,
+			25,
+			30, // 5
+			35,
+			40,
+			45,
+			50,
+			55, // 10
+			60,
+			65,
+			70,
+			75,
+			80, // 15
+			85,
+			90,
+			95,
+			100,
+			105, // 20
+			110,
+			115,
+			120,
+			125,
+			130, // 25
+			135,
+			140,
+			145,
+			150,
+			155, // 30
+			160,
+			165,
+			170,
+			175,
+			180, // 35
+			185,
+			190,
+			195,
+			200,
+			200, // 40
+			200,
+			200,
+			200,
+			200,
+			200, // 45
+			200,
+			200,
+			200,
+			200,
+			200, // 50
+			205,
+			210,
+			215,
+			220,
+			225, // 55
+			230,
+			235,
+			240,
+			245,
+			250, // 60
+			250,
+			250,
+			250,
+			250,
+			250, // 65
+			255,
+			260,
+			265,
+			270,
+			275, // 70
+	};
+
+	int EnhancedFixedWeaponSkillTable[] = {
```

Comment:
This one needs to be changed, too, but the math isn't exactly very formulaic. Are these numbers made up or actually pulled from somewhere? Would really prefer not to hard-code an array of ~70 numbers like this.

Reply:
They were to mimic how the old TAKP combat "felt", each level was tested to make them feel similar as the mobs on stock are just way too easy it seemed.

Reply:
So if it's just randomly generated I guess we can kind of fudge it a little to get it at least ~99% accurate using math, seems to be `(level * 6) + X` up to a certain point.

Reply:
What about this?

```cpp
	int  level = EQ::ClampUpper(std::max(1, static_cast<int>(GetLevel())), 70);
	bool UseEnhanced = RuleB(Combat, UseEnhancedMobStaticWeaponSkill);
	int  fixed_weapon_skill = 0;

	if (UseEnhanced) {
		if (level < 1) {
			fixed_weapon_skill = 5;
		} else if (level <= 20) {
			fixed_weapon_skill =  5 + (level - 1) * 6; // Rough approximation for levels 1 to 20
		} else if (level <= 40) {
			fixed_weapon_skill = 124 + (level - 20) * 6; // Rough approximation for levels 21 to 40
		} else if (level <= 60) {
			fixed_weapon_skill = 250 + (level - 40) * 2; // Rough approximation for levels 41 to 60
		} else {
			fixed_weapon_skill = 318 + (level - 60); // Rough approximation for levels 61 to 70
		}
	} else {
		fixed_weapon_skill = 275;

		if (level <= 39) {
			fixed_weapon_skill = (level * 5) + 5;
		} else if (EQ::ValueWithin(level, 40, 50)) {
			fixed_weapon_skill = 200;
		} else if (EQ::ValueWithin(level, 51, 60)) {
			fixed_weapon_skill = 200 + ((level - 50) * 5);
		} else if (EQ::ValueWithin(level, 61, 65)) {
			fixed_weapon_skill = 250;
		} else if (EQ::ValueWithin(level, 66, 70)) {
			fixed_weapon_skill = 250 + ((level - 65) * 5);
		}
	}

	return fixed_weapon_skill;
	```

Reply:
Well with the UseEnchanced logic actually flipped and correct :)

Reply:
Okay, here we go.
```cpp
int Mob::GetMobFixedWeaponSkill()
{
	// Due to new code using a combination of Offense and Weapon skill to determine hit, depending on the class
	// and weapon wielded by a mob, the hit rate of an equal level mob could vary between 15% and 60%, which made
	// many mobs far too easy.  This particular call replaces the weapon/class based Weapon Skill with a fixed value.
	// Two tables exist, one equal to a Warrior of appropriate level, and one modified to make hit rate equal to the old code
	// assuming the UseMobFixedOffenseSkill flag is set TRUE or the mob class is a Warrior (all the the bonus is in Weapon Skill).

	int level = EQ::ClampUpper(std::max(1, static_cast<int>(GetLevel())), 70);

	if (!RuleB(Combat, UseEnhancedMobStaticWeaponSkill)) {
		if (level <= 39) {
			return (level * 5) + 5;
		} else if (EQ::ValueWithin(level, 40, 50)) {
			return 200;
		} else if (EQ::ValueWithin(level, 51, 60)) {
			return 200 + ((level - 50) * 5);
		} else if (EQ::ValueWithin(level, 61, 65)) {
			return 250;
		}

		return 250 + ((level - 65) * 5);
	}

	if (level >= 1 && level <= 39) {
		return (level * 6) - 1;
	} else if (level >= 45 && level <= 49) {
		return 260;
	} else if (level >= 50 && level <= 54) {
		return (level * 6) + 1;
	} else if (level >= 55 && level <= 59) {
		return (level * 7) + 5;
	} else if (level >= 60 && level <= 65) {
		return (level * 5) + 59;
	}

	return 330 + (level - 66);
}
```